src/protobuf-net.Reflection.Test/Parsers.cs

```diff
@@ -19,5 +19,22 @@ public void EmbeddedImportsWork()
             var err = set.GetErrors();
             Assert.Empty(err);
         }
+
+        [Fact]
+        public void BuildTypeHierarchyForServices()
+        {
+
+            var set = new FileDescriptorSet();
+            using (var file = File.OpenText("basic_service.proto"))
+            {
+                Assert.True(set.Add("basic_service.proto", true, file));
+            }
+            set.Process();
+            var err = set.GetErrors();
+            Assert.Empty(err);
+
+            Assert.Equal(".HelloWorld.HelloService", set.Files[0].Services[0].FullyQualifiedName);
```

Comment:
@mgravell any reason for the `.` prefix on FQN? It's this line that adds the dot:
https://github.com/protobuf-net/protobuf-net/blob/706684b8b56dc54f9a03b1d44a4f85a750911243/src/protobuf-net.Reflection/Parsers.cs#L1028

Reply:
because in .proto terms, `.foo.bar.blah` is an absolute path; `foo.bar.blah` is a *relative* path, which may resolve to the root, or may resolve to something else; in C# terms, this is the same as `SomeType` vs `global::SomeType`

Reply:
Ah, thanks for the clarification! Had never seend the `.` prefix before, but makes sense.